frontend/src/citizen-frontend/calendar/HolidayModal.tsx

```diff
@@ -46,26 +141,72 @@ export const HolidayModal = React.memo(function HolidayModal({
         holidayPeriods,
         ([holidayPeriod]) =>
           holidayPeriod && (
-            <>
-              <div>
+            <HolidayContainer>
+              <HolidaySection>
                 <div>{holidayPeriod.description[lang]}</div>
                 <ExternalLink
                   text={i18n.calendar.holidayModal.additionalInformation}
                   href={holidayPeriod.descriptionLink[lang]}
                   newTab
                 />
-              </div>
-              {availableChildren
-                .map(
-                  (child) =>
-                    child.preferredName || child.firstName.split(' ')[0]
+              </HolidaySection>
+              {Object.keys(form.selections).map((childId) => {
+                const child = childWithId(childId)
+                return (
+                  child && (
+                    <HolidaySection key={childId}>
+                      <HolidaySelector
+                        key={childId}
+                        child={child}
+                        freeAbsencePeriod={holidayPeriod?.freePeriod}
+                        value={form.selections[childId].selectedFreePeriod}
+                        onSelectPeriod={(dateRange) =>
+                          selectFreePeriod(childId, dateRange)
+                        }
+                      />
+                    </HolidaySection>
+                  )
                 )
-                .join(', ')}
-            </>
+              })}
+            </HolidayContainer>
           )
       )}
     </AsyncFormModal>
   )
 })
 
+const HolidayContainer = styled(FixedSpaceColumn)`
+  gap: ${defaultMargins.s};
+`
```

Comment:
Mixing `FixedSpaceColumn` and `gap` is redundant because `FixedSpaceColumn` tries to do the same thing that `gap` does.

Reply:
TIL :D